code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,468 @@
+		update_icon()
+
+	if(powernet && input_attempt && turned_on)
+		input_pulling = min(input_available, input_number * power_format_multi)
+
+		if(inputting)
+			if(input_pulling > 0)
+				consume_direct_power(input_pulling)
+				charge += input_pulling
+			else
+				inputting = FALSE
+		else
+			if(input_attempt && input_pulling > 0)
+				inputting = TRUE
+	else
+		inputting = FALSE
+
+	if(charge < MINIMUM_POWER)
+		firing = FALSE
+		output_level = 0
+		destroy_lasers()
+		return
+
+	if(!firing)
+		return
+
+	output_level = min(charge, output_number * power_format_multi_output)
+
+	sell_power(output_level * WATT_TICK_TO_JOULE)
+
+	charge -= output_level
+
+//// Selling defines are here
+#define MINIMUM_BAR 1
+#define PROCESS_CAP (20 - MINIMUM_BAR)
+
+#define A1_CURVE 20
+
+#define HIGH_CUT_RATIO 0.75
+#define MEDIUM_CUT_RATIO 0.25
+
+/obj/machinery/power/transmission_laser/proc/sell_power(joules)
+	var/mega_joules = joules / (1 MW)
+
+	var/generated_cash = (2 * mega_joules * PROCESS_CAP) / ((2 * mega_joules) + (PROCESS_CAP * A1_CURVE))
+	generated_cash += (4 * mega_joules * MINIMUM_BAR) / (4 * mega_joules + MINIMUM_BAR)
+	if(generated_cash < 0)
+		return
+
+	total_energy += joules
+	total_earnings += generated_cash
+	unsent_earnings += generated_cash
+
+	var/datum/money_account/engineering_bank_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_ENGINEERING)
+	var/datum/money_account/cargo_bank_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_SUPPLY)
+
+	if(unsent_earnings > 200)
+		var/medium_cut = round(unsent_earnings * MEDIUM_CUT_RATIO)
+		var/high_cut = round(unsent_earnings * HIGH_CUT_RATIO)
+
+		GLOB.station_money_database.credit_account(cargo_bank_account, medium_cut, "Transmission Laser Payout", "Central Command Supply Master", supress_log = FALSE)
+		unsent_earnings -= medium_cut
+
+		GLOB.station_money_database.credit_account(engineering_bank_account, high_cut, "Transmission Laser Payout", "Central Command Supply Master", supress_log = FALSE)
+		unsent_earnings -= high_cut
+
+#undef A1_CURVE
+#undef PROCESS_CAP
+#undef MINIMUM_BAR
+#undef HIGH_CUT_RATIO
+#undef MEDIUM_CUT_RATIO
+
+// Beam related procs
+
+/obj/machinery/power/transmission_laser/proc/setup_lasers()
+	if(charge < 1 MJ) // We don't have enough power to setup the beam and we aren't receiving any.
+		return
+	var/turf/last_step = get_step(get_front_turf(), dir)
+	for(var/num in 1 to range)
+		if(!(locate(/obj/effect/transmission_beam) in last_step))
+			var/obj/effect/transmission_beam/new_beam = new(last_step, src)
+			new_beam.host = src
+			new_beam.dir = dir
+			laser_effects += new_beam
+
+		last_step = get_step(last_step, dir)
+
+/obj/machinery/power/transmission_laser/proc/destroy_lasers()
+	for(var/obj/effect/transmission_beam/listed_beam as anything in laser_effects)
+		laser_effects -= listed_beam
+		qdel(listed_beam)
+
+// Beam
+/obj/effect/transmission_beam
```

Comment:
It cant be a /obj/effect/ebeam for what you want to do with it?

Reply:
No. Following a discussion with balance and design the beam is going to be a purely visual effect that goes under everything. Making it an ebeam will only make it more complicated.